contrib/win32/win32compat/tncon.c

```diff
@@ -173,17 +174,8 @@ ReadConsoleForTermEmul(HANDLE hInput, char *destin, int destinlen)
 				break;
 
 			case KEY_EVENT:
-				bCapsOn = (inputRecord.Event.KeyEvent.dwControlKeyState & CAPSLOCK_ON);
-				bShift = (inputRecord.Event.KeyEvent.dwControlKeyState & SHIFT_PRESSED);
-				dwControlKeyState = inputRecord.Event.KeyEvent.dwControlKeyState &
-					~(CAPSLOCK_ON | ENHANCED_KEY | NUMLOCK_ON | SCROLLLOCK_ON);
-
-				/* ignore the AltGr flags*/
-				if ((dwControlKeyState & dwAltGrFlags) == dwAltGrFlags)
-					dwControlKeyState = dwControlKeyState & ~dwAltGrFlags;
-
-				modKey = GetModifierKey(dwControlKeyState);
-				if (inputRecord.Event.KeyEvent.bKeyDown) {
+				if ((inputRecord.Event.KeyEvent.bKeyDown && wcscmp(&inputRecord.Event.KeyEvent.uChar.UnicodeChar, L"")) ||
```

Comment:
this is _very dangerous_. wcscmp is uncounted, and will run off the end of the buffer that contains UnicodeChar. Why should we not just use `...UnicodeChar == L'\0'`?